mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -212,12 +230,140 @@ class LinearRegressionModel private[ml] (
   extends RegressionModel[Vector, LinearRegressionModel]
   with LinearRegressionParams {
 
+  private var trainingSummary: Option[LinearRegressionTrainingSummary] = None
```

Comment:
I see; I thought the use case was for sending stats to workers.  For model save/load, I agree we will need to save local stats by default.  I'm not sure that marking items as transient is the way to do that, unless we want to encourage users to save models as POJOs (which we should not).  I think that a follow-up PR should add the summary to the saved model, where it saves only the local values.  When the model is loaded from disk, the values not saved can be set to null.  Is there a need to mark things as transient?